core/blockchain.go

```diff
@@ -932,7 +932,7 @@ func (bc *BlockChain) WriteBlockWithState(block *types.Block, receipts []*types.
 					}
 				}
 				// If optimum or critical limits reached, write to disk
-				if chosen >= lastWrite+triesInMemory || size >= 2*limit || bc.gcproc >= 2*bc.cacheConfig.TrieTimeLimit {
+				if chosen >= lastWrite+triesInMemory || size >= limit || bc.gcproc >= bc.cacheConfig.TrieTimeLimit {
```

Comment:
The old code is correct here.

`size > limit || bc.gcproc > bc.cacheConfig.TrieTimeLimit` already holds true from line 923. That clause should only cause a commit if `chosen >= lastWrite+triesInMemory` is also true to avoid writing too much to disk.

The `size >= 2*limit || bc.gcproc >= 2*bc.cacheConfig.TrieTimeLimit` clauses ensure that is our machine is too slow to commit to meat the `chosen >= lastWrite+triesInMemory` base requirements, then we have a bailout clause to prevent crashing with oom.